country_plan/models.py

```diff
@@ -0,0 +1,105 @@
+
+
+def pdf_upload_to(instance, filename):
+    date_str = timezone.now().strftime('%Y-%m-%d-%H-%M-%S')
+    return f'country-plan-pdf/{date_str}/{filename}'
+
+
+class CountryPlanBase(models.Model):
+    created_by = models.ForeignKey(
+        User,
+        verbose_name=_('Created By'),
+        blank=True,
+        null=True,
+        related_name='data_import_created_by',
+        on_delete=models.SET_NULL,
+    )
+    created_at = models.DateTimeField(verbose_name=_('Created at'), auto_now_add=True)
+    updated_by = models.ForeignKey(
+        User,
+        verbose_name=_('Updated by'),
+        blank=True,
+        null=True,
+        related_name='data_import_updated_by',
+        on_delete=models.SET_NULL
+    )
+    updated_at = models.DateTimeField(verbose_name=_('Updated at'), auto_now=True)
+
+
+class DataImport(CountryPlanBase):
+    file = models.FileField(verbose_name=_('EXEL file'), upload_to=file_upload_to)
+
+    def __str__(self):
+        return self.file.name
+
+
+class CountryPlan(CountryPlanBase):
+    internal_plan_file = models.FileField(verbose_name=_('Internal Plan'), upload_to=pdf_upload_to)
+    public_plan_file = models.FileField(verbose_name=_('Country Plan'), upload_to=pdf_upload_to)
+    country = models.ForeignKey(Country, on_delete=models.SET_NULL, null=True, blank=True)
+    requested_amount = models.IntegerField(verbose_name=_('Requested Amount'), blank=True, null=True)
+    people_targeted = models.IntegerField(verbose_name=_('People Targeted'), blank=True, null=True)
+    situation_analysis = models.TextField(blank=True, verbose_name=_('Situation Analysis'))
+    role_of_national_society = models.TextField(blank=True, verbose_name=_('Role of National Society'))
+    is_publish = models.BooleanField(default=False)
+
+    def __str__(self):
+        return f'{self.country}'
+
+
+class StrategicPriority(models.Model):
+    class StrategicPriorityName(models.TextChoices):
+        SP_1 = 'sp_1', _('Climate and environmental crisis')
+        SP_2 = 'sp_2', _('Evolving crisis and disasters')
+        SP_3 = 'sp_3', _('Growing gaps in health and wellbeing')
+        SP_4 = 'sp_4', _('Migration and Identity')
+        SP_5 = 'sp_5', _('Value power and inclusion')
+
+    country_plan = models.ForeignKey(
+        CountryPlan, on_delete=models.CASCADE,
+        verbose_name=_('Country Plan'),
+        related_name='country_plan_sp'
+    )
+    sp_name = models.CharField(
+        max_length=100, choices=StrategicPriorityName.choices,
+        null=True, blank=True, verbose_name=_('share with')
+    )
+    funding_requirement = models.IntegerField(verbose_name=_('Funding Requirement'), blank=True, null=True)
+    people_targeted = models.IntegerField(verbose_name=_('People Targeted'), blank=True, null=True)
+
+    def __str__(self):
+        return f'{self.country_plan.iso3}'
+
+
+class MembershipCoordination(models.Model):
+    class MembershipCoordinationChoices(models.TextChoices):
```

Comment:
Same here, maybe use Type as well.
Also, let's not add the `Choices` postfix and class name as prefixes either. Since we use it with the class `MembershipCoordination.MembershipCoordinationChoices.choices`, it looks repetitive.
```suggestion
    class Type(models.TextChoices):
```